release-note-script/src/main/java/MergeReleaseNotes.java

```diff
@@ -0,0 +1,240 @@
+ * release note bodies for ScalarDB, ScalarDB Cluster, ScalarDB GraphQL, and ScalarDB SQL as
+ * markdown files as its input. And then output the merged release note body to standard output as a
+ * markdown.
+ *
+ * <p>This script is assumed to be executed in a GitHub Actions workflow.
+ *
+ * <p>Note that it is needed that java 11 to execute this script since it is executed as a
+ * Single-File Source-code program.
+ */
+@SuppressWarnings("DefaultPackage")
+public class MergeReleaseNotes {
+
+  private static final String DEBUG = System.getenv("DEBUG");
+  private static final String SECTION_SUMMARY = "Summary";
+
+  private final Map<Edition, Map<Category, Map<Repository, List<ReleaseNote>>>> editionMap =
+      new EnumMap<>(Edition.class);
+
+  public static void main(String... args) throws Exception {
+    if (args.length == 0) {
+      System.err.printf(
+          "Usage: java --source 11 %s.java scalardb.md cluster.md graphql.md sql.md%n",
+          MergeReleaseNotes.class.getSimpleName());
+    }
+
+    MergeReleaseNotes mergeReleaseNotes = new MergeReleaseNotes();
+    mergeReleaseNotes.createMergedReleaseNote(args);
+  }
+
+  public void createMergedReleaseNote(String... args) throws Exception {
+    for (String path : args) {
+      Edition edition;
+      Repository repository;
+
+      switch (path) {
+        case "scalardb.md":
+          edition = Edition.COMMUNITY;
+          repository = Repository.DB;
+          break;
+        case "cluster.md":
+          edition = Edition.ENTERPRISE;
+          repository = Repository.CLUSTER;
+          break;
+        case "graphql.md":
+          edition = Edition.ENTERPRISE;
+          repository = Repository.GRAPHQL;
+          break;
+        case "sql.md":
+          edition = Edition.ENTERPRISE;
+          repository = Repository.SQL;
+          break;
+        default:
+          throw new IllegalArgumentException("Unknown file name: " + path);
+      }
+
+      File file = new File(path);
+      load(edition, repository, file);
+    }
+    output();
+  }
+
+  void load(Edition edition, Repository repository, File file) throws Exception {
+    Category category = null;
+    String line;
+    Matcher matcher;
+
+    try (BufferedReader reader = new BufferedReader(new FileReader(file, StandardCharsets.UTF_8))) {
+      while ((line = reader.readLine()) != null) {
+        if (DEBUG != null) System.err.printf("ReadLine: %s%n", line);
+        matcher = Pattern.compile("^## *(\\p{Print}+) *$").matcher(line);
+        if (matcher.matches()) {
+          if (DEBUG != null) System.err.printf("matched: %s%n", matcher.group(1));
+          if (!SECTION_SUMMARY.equalsIgnoreCase(matcher.group(1)))
+            category = Category.getByDisplayName(matcher.group(1));
+          continue;
+        }
+
+        matcher = Pattern.compile("^ *- *(\\p{Print}+)$").matcher(line);
+        if (matcher.matches()) {
+          if (category == null)
```

Comment:
> Once a category is set with the first pattern match, it seems that the category will never be unset.

Category will be reset when `reader.readLine()` read the header line.

For example, this script reads the release note file as like [this](https://github.com/scalar-labs/actions/blob/add-releasenote-script/release-note-script/src/test/resources/scalardb.md). When it read the each h2 header (e.g. `## Summary`, `## Enhancements`, etc.) except for summary header (i.e. `## Summary`), it handle as the start of a category section. So, it will change the value of `category` variable when it reads the header line ([at this point](https://github.com/scalar-labs/actions/blob/add-releasenote-script/release-note-script/src/main/java/MergeReleaseNotes.java#L88)).